packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -32,17 +26,15 @@ abstract class LayoutComponent extends PositionComponent {
     layoutChildren();
   }
 
-  double get gap => _gap;
-
-  NotifyingVector2? get _bounds =>
-      isManuallySized ? size : findParent<PositionComponent>()?.size;
-
   @override
-  Future<void> onLoad() async {
-    final size = _bounds;
-    size?.addListener(layoutChildren);
+  set size(Vector2 size) {
+    if (_allowSetSize) {
+      this.size = size;
+    }
```

Comment:
otherwise, some kind of error should be reported to the user, right?

Reply:
By throwing an exception or how is this normally done within Flame?

Reply:
I think an assertion is fine